### PR TITLE
improve ci

### PR DESCRIPTION
**Pull request by ccoVeille**
- **ci: bump golangci-lint to latest version**
- **chore: fix golangci-lint errors**
- **ci: enable mirror linter and fix reported errors**
- **ci: enable errorllint and fix reported errors**
- **ci: enable thelper and fix reported errors**
- **ci: enable usestdlibvars and fix reported errors**
- **ci: enable revive.bool-literal-in-expr and fix reported errors**
- **ci: enable revive.comment-spacing and fix reported errors**
- **ci: enable revive.bare-return and fix reported errors**
- **ci: enable revive.early-return and fix reported errors**
- **ci: enable revive.defer and fix reported errors**

Related to #787

### CODE REVIEW

**Review by iamgoroot — APPROVED** [review]
nice!

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
-    directory: "/" 
+  - package-ecosystem: "gomod"
+    open-pull-requests-limit: 10  # avoid spam, if no one reacts
+    directory: "/"
     schedule:
       interval: "weekly"
+    groups:
+      all:
+        patterns:
+          - "*"  # Group all updates into a single larger pull request.
```
[the remark this answers is not in the document]
Comment (ccoVeille) [reply .github/dependabot.yml:12]: OK, I will check if I can split by minor patch

I assume patch should pass.

Please confirm

[the remark this answers is not in the document]
Comment (aryanmehrotra) [reply .github/dependabot.yml:12]: Lets keep all of them separate as based on this PR, where redis has changed support in minor release - https://github.com/gofr-dev/gofr/pull/702

[the remark this answers is not in the document]
Comment (ccoVeille) [reply .github/dependabot.yml:12]: OK indeed

[the remark this answers is not in the document]
Comment (ccoVeille) [reply .github/dependabot.yml]: The commit has been removed from this PR, the following one has been opened to address the github-actions updates

- #792

File: pkg/gofr/handler.go [hunk 2]
```diff
@@ -126,13 +128,14 @@ func (h handler) setContextTimeout(timeout string) int {
...
 
-func panicRecoveryHandler(log logging.Logger, panicked chan struct{}) {
-	re := recover()
-	if re != nil {
-		close(panicked)
-		log.Error(panicLog{
-			Error:      fmt.Sprint(re),
-			StackTrace: string(debug.Stack()),
-		})
+func panicRecoveryHandler(re any, log logging.Logger, panicked chan struct{}) {
```
Comment (vipul-rawat) [inline pkg/gofr/handler.go:131]: @ccoVeille why not call the `recover()` in panicRecoveryHandler instead of passing re!

Comment (ccoVeille) [reply pkg/gofr/handler.go:131]: Honestly, I wonder too

That's the only way I found to cimply with the revive.defer linter.

Adding a //nolint to the function was not enough.

Maybe there is something I missed. Someone might find a solution, but for now. The recover call is clear now.

Comment (vipul-rawat) [reply pkg/gofr/handler.go:131]: Yeah, not sure why the linter is doing it, I think we can change it later if we can find a solution to this.

**Review by vipul-rawat — APPROVED** [review]

**Review by Umang01-hash — APPROVED** [review]

**Review by aryanmehrotra — APPROVED** [review]

**Pull request merged**